unstructured/ingest/connector/weaviate.py

```diff
@@ -0,0 +1,134 @@
+import json
+import typing as t
+from dataclasses import dataclass
+
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+
+@dataclass
+class SimpleWeaviateConfig(BaseConnectorConfig):
+    host_url: str
+    class_name: str
+    auth_keys: t.Optional[t.Dict[str, str]] = None
+
+
+@dataclass
+class WeaviateWriteConfig(WriteConfig):
+    batch_size: int = 100
+
+
+@dataclass
+class WeaviateDestinationConnector(BaseDestinationConnector):
+    write_config: WeaviateWriteConfig
+    connector_config: SimpleWeaviateConfig
+
+    @requires_dependencies(["weaviate"], extras="weaviate")
+    def initialize(self):
+        from weaviate import Client
+
+        auth = self._resolve_auth_method()
+
+        self.client: Client = Client(
```

Comment:
Might be good to log a [destination connection error](https://github.com/Unstructured-IO/unstructured/blob/9fd77a523207fe7845ede28444ed361784e7783e/unstructured/ingest/error.py#L36-L37) if it fails